bloom-housing/bloom
# 2818/existing user updates

## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/2818)


## Description
This changes the behavior when trying to create a user that already exists. Now if:
1) There is an existing partner xyz
2) Someone tries to create a new partner with xyz's email address but with a new jurisdiction
3) we send xyz an email saying they received updates, and update their user record to have the union between the jurisdictions and listing assignments for xyz's existing records and the new incoming ones
4) if the email existing, and (the existing user/the "new" user is not to a partner, or there is no jurisdictional mismatch) we still provide the "user already exists" warning

## How Can This Be Tested/Reviewed?
Create a new partner user, and assign them to some (but not) jurisdictions
move to create a new user with the same email as the newly created user, and give them a new jurisdiction

the user should receive an email saying they had updates, and the user's record in the partner portal should represent the union of jurisdictions and listing assignments

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
